contracts/utils/cryptography/ERC7913Utils.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.24;
+
+import {SignatureChecker} from "@openzeppelin/contracts/utils/cryptography/SignatureChecker.sol";
+import {Bytes} from "@openzeppelin/contracts/utils/Bytes.sol";
+import {IERC7913SignatureVerifier} from "../../interfaces/IERC7913.sol";
+
+/**
+ * @dev Helper library to verify key signatures following the ERC-7913 standard, with fallback to ECDSA and ERC-1271
+ * when the signer's key is empty (as specified in ERC-7913)
+ */
+library ERC7913Utils {
```

Comment:
I think I would prefer having just one implementation of the `isValidSignatureNow` function. The rationale is that `signature` always ends up in memory for any valid verification, and a portion of `signer` will be copied too. So I doubt there's actual benefits in having two versions. I'll simplify for now following #109

Reply:
The memory version uses a slice, that does a memory copy. 

Overall, calldata versions are often cheaper because they don't copy to memory until the very last point (when the Abi call is encoded) contrary to the memory version that copy to memory at least once more (in this case 2 times)

Overall not a big deal. Having a single "memory" version is probably good enough. Just explaining the logic

Reply:
Yeah I saw the memory version is required, so I'd agree with just one "memory" version for now unless there are significant savings measured.

Thanks for answering though, now please you go 👶🏻